main.py

```diff
@@ -15,22 +16,30 @@
 
 update_time = 10
 
-
-def check_letter(hand_landmarks):
+def get_letter(hand_landmarks):
     # Predict the sign letter in the image
     preprocessed_input = preprocess_keypoints(hand_landmarks)
     prediction = model.predict([preprocessed_input])
     predicted_letter = labels[np.argmax(prediction)]
 
+    return predicted_letter
+
+
+def check_letter(hand_landmarks):
     # Compare the predicted letter with the suggested letter
+    predicted_letter = get_letter(hand_landmarks)
     print(f'Current letter: {assistant.current_letter}. Predicted letter: {predicted_letter}')
+    
     return predicted_letter == assistant.current_letter
 
 
 def run():
     assistant.welcome()
     # For webcam input
     camera = cv2.VideoCapture(0)
+    user_attempts = {}
+    username = input("Enter username: ")
```

Comment:
Man skulle kunna ha den här inputen innan assistant.welcome() så får man börja med att skriva in namnet och sen får man välkomstmeddelandet